### PR TITLE
Consider java lang while forming GEN_STRING with validation enabled

### PR DESCRIPTION
**Pull request by venkatduddu**

### CODE REVIEW

File: all/entrypoint.sh [hunk 1]
```diff
@@ -396,6 +396,10 @@ if [[ $GEN_VALIDATOR == true && $GEN_LANG == "gogo" ]]; then
     GEN_STRING="$GEN_STRING --validate_out=lang=gogo${VALIDATOR_SOURCE_RELATIVE}:$OUT_DIR"
 fi
 
+if [[ $GEN_VALIDATOR == true && $GEN_LANG == "java" ]]; then
+    GEN_STRING="$GEN_STRING --validate_out=lang=java${VALIDATOR_SOURCE_RELATIVE}:$OUT_DIR"
```
Comment (ido-namely) [inline all/entrypoint.sh:400]: VALIDATOR_SOURCE_RELATIVE is a go/gogo flag, so there is no need to set it.

**Review by ido-namely — COMMENTED** [review]
Thanks for the pr @venkatduddu!
It would be great if you can add a test for this use case.
You can find our tests in https://github.com/namely/docker-protoc/blob/master/all/test.sh

File: all/entrypoint.sh (hunk 1 above)
Comment (venkatduddu) [reply all/entrypoint.sh]: @ido-namely took care of it

**Comment by venkatduddu** [comment]
> Thanks for the pr @venkatduddu! It would be great if you can add a test for this use case. You can find our tests in https://github.com/namely/docker-protoc/blob/master/all/test.sh

@ido-namely tried adding a test for java `--with-validator` flag, however it wasn't generating any validation related java classes/methods. Do I need to modify the test.proto to include some validations?

**Comment by ido-namely** [comment]
> > Thanks for the pr @venkatduddu! It would be great if you can add a test for this use case. You can find our tests in https://github.com/namely/docker-protoc/blob/master/all/test.sh
> 
> @ido-namely tried adding a test for java `--with-validator` flag, however it wasn't generating any validation related java classes/methods. Do I need to modify the test.proto to include some validations?
> 
> image

Hi @venkatduddu , that would be my guess, perhaps in Java it does not generate an additional file if there aren't any annotations.

**Comment by venkatduddu** [comment]
> > > Thanks for the pr @venkatduddu! It would be great if you can add a test for this use case. You can find our tests in https://github.com/namely/docker-protoc/blob/master/all/test.sh
> > 
> > 
> > @ido-namely tried adding a test for java `--with-validator` flag, however it wasn't generating any validation related java classes/methods. Do I need to modify the test.proto to include some validations?
> > image
> 
> Hi @venkatduddu , that would be my guess, perhaps in Java it does not generate an additional file if there aren't any annotations.

Hi @ido-namely you are right, i added validate.proto and it generated the java validator.
Have a look and see how does it look.

File: all/test.sh [hunk 2]
```diff
@@ -187,9 +187,7 @@ testGeneration() {
         fi
     fi
 
-        # Test that we have generated the test.pb.go file.
```
Comment (ido-namely) [inline all/test.sh:190]: we'd want to keep this comment

File: all/test.sh [hunk 3]
```diff
@@ -207,6 +205,23 @@ testGeneration() {
         fi
     fi
 
+    if [[ "$extra_args" == *"--with-validator"* ]] && [[ "$lang" == "java" ]]; then
+        # Test that we have generated the Test.java files and TestValidator.java files
+        java_file_count=$(find $expected_output_dir -type f -name "Test.java" | wc -l)
+        if [ $java_file_count -ne 1 ]; then
+            echo >&2 "[Fail] $name"
+            echo >&2 "Test.java files were not generated in $expected_output_dir"
```
Comment (ido-namely) [inline all/test.sh:213]:
```suggestion
            echo >&2 "Test.java file was not generated in $expected_output_dir"
```

File: all/test.sh [hunk 4]
```diff
@@ -207,6 +205,23 @@ testGeneration() {
...
+            exit 1
+        fi
+        java_validator_file_count=$(find $expected_output_dir -type f -name "TestValidator.java" | wc -l)
+        if [ $java_validator_file_count -ne 1 ]; then
+            echo >&2 "[Fail] $name"
+            echo >&2 "TestValidator.java files were not generated in $expected_output_dir"
```
Comment (ido-namely) [inline all/test.sh:219]:
```suggestion
            echo >&2 "TestValidator.java file was not generated in $expected_output_dir"
```

File: all/test.sh [hunk 5]
```diff
@@ -207,6 +205,23 @@ testGeneration() {
...
+            exit 1
+        fi
+        echo "java with validator"
```
Comment (ido-namely) [inline all/test.sh:222]: this can be removed

File: all/test/test.proto [hunk 6]
```diff
@@ -5,6 +5,7 @@ package Messages;
 import "all/test/include.proto";
 import "google/api/annotations.proto";
 import "google/protobuf/field_mask.proto";
+import "all/test/validate.proto";
```
Comment (ido-namely) [inline all/test/test.proto:8]:
```suggestion
import "validate/validate.proto";
```

The file is included in the docker image (in /opt/include/validate/validate.proto) so you don't need to add it to the repo.

File: all/test.sh (hunk 2 above)
Comment (venkatduddu) [reply all/test.sh:190]: taken care of it

File: all/test.sh (hunk 3 above)
Comment (venkatduddu) [reply all/test.sh]: taken care of it

File: all/test.sh (hunk 4 above)
Comment (venkatduddu) [reply all/test.sh]: taken care of it

File: all/test/test.proto (hunk 6 above)
Comment (venkatduddu) [reply all/test/test.proto]: took care of it

**Review by ido-namely — APPROVED** [review]

**Review by abe545 — APPROVED** [review]

**Comment by ido-namely** [comment]
Now available in `v1.46_1`.
Thank you for your contribution @venkatduddu !

**Pull request merged**